### PR TITLE
Allow user to plot ridgeplots for specific precincts by passing in names of precincts

### PR DESCRIPTION
**Pull request by matthewkol186**
# Changes made
* By default, we show all precincts
* Adapt height of plot to number of precincts being shown
* Decompose `plot_precincts` (which was getting kind of big) into `plot_precincts` and `plot_single_ridgeplot`
* Add informative docstrings to `plot_precincts` and `plot_single_ridgeplot`
* Reduce the number of instance variables in `TwoByTwoEI` (because linter was complaining) by collapsing `self.sampled_voting_prefs_district_gp1 = None; self.sampled_voting_prefs_district_gp2 = None` into `self.sampled_voting_prefs = [None, None]` (same for `self.posterior_mean_voting_prefs_district_gp1/gp2` and `self.credible_interval_95_mean_voting_prefs_district_gp1/gp2`
* A "rider amendment" added along with this change: the box-and-whisker plots now show 95% credible intervals by placing the percentile range from 2.5 to 97.5 (rather than 5 to 95, which it was previously)

# Usage
```
precinct_names = ["P1", "P2", "P3"]
ei.fit(X, T, N, demographic_group_name=demographic_name, candidate_name=candidate_name, precinct_names=precinct_names)
...
ei.precinct_level_plot(precinct_names=["P1, "P2"])
# ridgeplots only shown for P1 and P2
```

# Before changes
image
Note: no y-axis label, each curve is not individually labeled on the y-axis

# After changes

## Default behavior (no specification of precinct names)
image

## Specifying precincts to be plotted (n=10, n=25, n=60)
image
image
image
**Note that we by default will only plot the first 50 precincts, even if they are specified by name by the user. The user has to specify `show_all_precincts=True` for us to show all of them.**

### CODE REVIEW

File: pyei/plot_utils.py [hunk 1]
```diff
@@ -83,7 +125,7 @@ def plot_boxplot(voting_prefs_group1, voting_prefs_group2, group1_name, group2_n
     if ax is None:
         ax = plt.gca()
     samples_df = pd.DataFrame({group1_name: voting_prefs_group1, group2_name: voting_prefs_group2})
-    ax = sns.boxplot(data=samples_df, orient="h", whis=[5, 95], ax=ax)
+    ax = sns.boxplot(data=samples_df, orient="h", whis=[2.5, 97.5], ax=ax)
```
Comment (matthewkol186) [inline pyei/plot_utils.py:128]: This is the first "rider change" where I realized that the whiskers were showing a 90% credible interval, not a 95% credible interval!

File: pyei/plot_utils.py [hunk 2]
```diff
@@ -99,7 +141,7 @@ def plot_summary(
     sns.despine(ax=ax_box, left=True)
     # plot custom boxplot, with two boxplots in the same row
     colors = sns.color_palette()  # fetch seaborn default color palette
-    plot_props = dict(fliersize=5, linewidth=2, whis=[5, 95])
+    plot_props = dict(fliersize=5, linewidth=2, whis=[2.5, 97.5])
```
Comment (matthewkol186) [inline pyei/plot_utils.py:144]: This is the second "rider change" where I realized that the whiskers were showing a 90% credible interval, not a 95% credible interval!

File: pyei/plot_utils.py (hunk 1 above)
Comment (karink520) [reply pyei/plot_utils.py:128]: Nice catch

**Comment by karink520** [comment]
Nice! This is a great feature in the plotting!

**Pull request merged**